### PR TITLE
chore: add details to complex prop type

### PR DESCRIPTION
**Pull request by rayhan-ab**
example of complex prop type is `SelectOption`. the goal of this PR is to give more detail about the prop in storybook docs, for example `SelectOption` is `{label: string, value: string}`

### CODE REVIEW

File: src/components/Page/ProgressModalPage.stories.tsx [hunk 1]
```diff
@@ -21,25 +21,31 @@ const Template: Story<ProgressModalPageProps> = (args) => {
 
   const clickPrev = () => {
     if (step === 1) return;
-    setStep((prev) => prev - 1);
+    setStep((prev) => Math.min(args.stepLabels.length, prev - 1));
```
Comment (aziz-zen-ab) [inline src/components/Page/ProgressModalPage.stories.tsx:24]: I don't know what happen but in my local the progress side is not working

File: src/components/styled-atlaskit/Dropdown/index.tsx [hunk 2]
```diff
@@ -5,5 +5,19 @@
...
+  DropdownItem as DItem,
+  DropdownItemGroup as DIG,
+  DropdownMenuStatefulProps,
+} from "@atlaskit/dropdown-menu";
+import { DropdownItemProps } from "@atlaskit/dropdown-menu/dist/cjs/components/item/DropdownItem";
+import { Props } from "@atlaskit/dropdown-menu/dist/cjs/components/group/DropdownItemGroup";
+
+// doing this only to populate all props and their description in storybook docs
+// otherwise we can simply export { default as DropdownMenu, DropdownItem, DropdownItemGroup } from "@atlaskit/dropdown-menu"
+export const DropdownMenu = (props: DropdownMenuStatefulProps) => <DM {...props} />;
```
Comment (aziz-zen-ab) [inline src/components/styled-atlaskit/Dropdown/index.tsx:20]: have you check the implementation in AP? Sometime the mandatory props have to stated every time we use this component.

**Review by aziz-zen-ab — COMMENTED** [review]
Some components are updated their documentation comment, others need some adjustment. Don't you think it better to update the Changelog.md also?

File: src/components/styled-atlaskit/Dropdown/index.tsx (hunk 2 above)
Comment (rayhan-ab) [reply src/components/styled-atlaskit/Dropdown/index.tsx:20]: what do you mean?

File: src/components/Page/ProgressModalPage.stories.tsx (hunk 1 above)
Comment (rayhan-ab) [reply src/components/Page/ProgressModalPage.stories.tsx:24]: try again

File: src/components/styled-atlaskit/Dropdown/index.tsx (hunk 2 above)
Comment (rayhan-ab) [reply src/components/styled-atlaskit/Dropdown/index.tsx:20]: i think i know what you mean. turns out a lot of props are mandatory

File: src/components/Page/ProgressModalPage.stories.tsx (hunk 1 above)
Comment (aziz-zen-ab) [reply src/components/Page/ProgressModalPage.stories.tsx:24]: lgtm

File: src/components/styled-atlaskit/Dropdown/index.tsx (hunk 2 above)
Comment (rayhan-ab) [reply src/components/styled-atlaskit/Dropdown/index.tsx]: i turn the props to Partial

**Review by aziz-zen-ab — APPROVED** [review]

**Pull request merged**